tests/mapper/test_graph_generator.py

```diff
@@ -0,0 +1,138 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from stellargraph.core.graph import *
+from stellargraph.mapper.graph_generator import GraphGenerator, GraphSequence
+
+import numpy as np
+import pytest
+from ..test_utils.graphs import example_graph_random, example_hin_1
+
+
+class Test_GraphGenerator:
+
+    graphs = [
+        example_graph_random(feature_size=4, n_nodes=6),
+        example_graph_random(feature_size=4, n_nodes=5),
+        example_graph_random(feature_size=4, n_nodes=3),
+    ]
+
+    graphs_nx = [
+        example_graph_random(feature_size=4, n_nodes=3, is_directed=False),
+        example_graph_random(
+            feature_size=4, n_nodes=2, is_directed=False
+        ).to_networkx(),
+    ]
+
+    graphs_diff_num_features = [
+        example_graph_random(feature_size=2, n_nodes=6),
+        example_graph_random(feature_size=4, n_nodes=5),
+    ]
+
+    graphs_mixed = [
+        example_graph_random(feature_size=2, n_nodes=6),
+        example_hin_1(is_directed=False),
+    ]
+
+    def test_generator_init(self):
+        generator = GraphGenerator(graphs=self.graphs)
+        assert len(generator.graphs) == len(self.graphs)
+
+    def test_generator_init_different_feature_numbers(self):
+        with pytest.raises(ValueError):
+            generator = GraphGenerator(graphs=self.graphs_diff_num_features)
+
+    def test_generator_init_nx_graph(self):
+        with pytest.raises(TypeError):
+            generator = GraphGenerator(graphs=self.graphs_nx)
+
+    def test_generator_init_hin(self):
+        with pytest.raises(ValueError):
+            generator = GraphGenerator(graphs=self.graphs_mixed)
+
+    def test_generator_flow_invalid_batch_size(self):
+        with pytest.raises(ValueError):
+            GraphGenerator(graphs=self.graphs).flow(graph_ilocs=[0], batch_size=-1)
+
+        with pytest.raises(TypeError):
+            GraphGenerator(graphs=self.graphs).flow(graph_ilocs=[0], batch_size=2.0)
+
+        with pytest.raises(ValueError):
+            GraphGenerator(graphs=self.graphs).flow(graph_ilocs=[0], batch_size=0)
+
+    def test_generator_flow_incorrect_targets(self):
+
+        generator = GraphGenerator(graphs=self.graphs)
+
+        with pytest.raises(ValueError):
+            generator.flow(graph_ilocs=[0, 1], targets=np.array([0]))
+
+        with pytest.raises(TypeError):
+            generator.flow(graph_ilocs=[0, 1], targets=1)
+
+    def test_generator_flow_no_targets(self):
+
+        generator = GraphGenerator(graphs=self.graphs)
+
+        seq = generator.flow(graph_ilocs=[0, 1, 2], batch_size=2)
+        assert isinstance(seq, GraphSequence)
+
+        assert len(seq) == 2  # two batches
+
+        # The first batch should be size 2 and the second batch size 1
+        batch_0 = seq[0]
+        assert batch_0[0][0].shape[0] == 2
```

Comment:
If you wanted, you could reduce how much code there is to maintain here by incorporating `test_generator_flow_check_padding` into this one. Specifically, instead of just checking one element of each shape, check the whole tuple, like `.shape == (2, 6, ...)` here and `.shape == (2, 3, ...)` for `batch_1`. 

In addition to reducing the code, I think this would also be good as check that the tensors are in the expected order, because we'll be checking the whole shapes and so capture the differences between the feature and adjacency matrices too.

Reply:
I can't compare the entire shape because of the random shuffling of the graphs when `on_epoch_end` is called; there is no guarantee of the size for the second item in the shape of the arrays. This is because the latter is set based on the largest graph in the batch and since the first batch has 2 graphs we don't know which two of sizes 6, 5, or 3 are in the batch. Essentially, the second value in batch_0[0][0].shape could be either 6 or 5. Similarly, for batch_1 we don't know which of the 3 graphs was left over for the second batch so it could be any of 6, 5, or 3 depending on which two are in the first batch.

Reply:
Also, I prefer to have a separate test for padding because then it is clear what the test is for and also I use a different sequence generator with 2 graphs with guaranteed maximum size. Although, in this case I can check the full shapes so I will update accordingly.